### ISSUE TITLE
Quarkus [1.13.3.Final] Classloader issue - works in QuarkusTest and not in quarkusDev. (Can't find class)

### ISSUE THREAD

**Issue by rroopreddy** [datastax/cassandra-quarkus#171]
Quarkus team recommended to enter the bug here.
Details with duplicatable project is included here: https://github.com/quarkusio/quarkus/issues/17197 

------
Quarkus [1.13.3.Final] Classloader issue - works in QuarkusTest and not in quarkusDev.

It works in test mode and not in quarkusDev mode. Seems like the System.getClassLoader() is different @QuarkusTest vs gradle quarkusDev. Class.forName(className) is returning null for a class (JksSslEngineFactory ) specified for SSL configuration. Tried to attach @ApplicationScoped to JksSslEngineFactory and various other approaches without much luck.

------
I think this is an issue with https://github.com/datastax/cassandra-quarkus

Basically it is not using the correct ClassLoader, instead of using the TCCL it appears to be using its own ClassLoader. I think the fix may be to add a getClassLoader method to com.datastax.oss.quarkus.runtime.internal.context.QuarkusDriverContext that always returns the TCCL, but I don't know enough about the extension to say for sure. Can you open an issue in their issue tracker?

**Comment by adutra** [comment]
Hi, thanks for reporting this.

The reasons why the driver doesn't use TCCL by default are explained [here](https://docs.datastax.com/en/drivers/java/4.11/com/datastax/oss/driver/api/core/session/SessionBuilder.html#withClassLoader-java.lang.ClassLoader-).

That said, if the Quarkus team says the TCCL should always be used, then I'm OK implementing the suggested changes.

Could you open a PR? The change IMO would be a one-liner; just change [this line](https://github.com/datastax/cassandra-quarkus/blob/a84b7a0f27562d2ccf30f30ceb7c0b305eaa25e8/runtime/src/main/java/com/datastax/oss/quarkus/runtime/internal/quarkus/CassandraClientProducer.java#L75) to:

```
        new QuarkusCqlSessionBuilder()
            .withConfigLoader(configLoaderBuilder.build())
            .withClassLoader(Thread.currentThread().getContextClassLoader());
```

If possible, add also an integration test that reproduces the issue (with a dummy subclass of `DefaultSslEngineFactory`).

**Comment by rroopreddy** [comment]
All the details are here: https://github.com/quarkusio/quarkus/issues/17197

Added the ZIP file with full project to duplicate the issue

[test-project.zip](https://github.com/datastax/cassandra-quarkus/files/6502526/test-project.zip)
